docs/developer/getting-started/development.md

```diff
@@ -76,8 +76,24 @@ Spoofed Types, like virtual types, add menu items but also define a spoofed sche
 
 > Any resources returned by `getInstances` should have a `kind` matching required type. This results in the tables showing the correct actions, handling create/edit, etc.
 
-### Proxy Object and Common Functionality
-When resources are retrieved from the store they will be wrapped in a Proxy object - `/plugins/steve/resource-proxy.js`. This exposes common properties and functions from `/plugins/steve/resource-instance.js`. These can be overridden per resource type via optional files in `/models`. For example the `nameDisplay` value for the type `management.cattle.io.user` avoids using the `nameDisplay` from `resource-instance` by adding a `nameDisplay` function to `/models/management.cattle.io.user.js`.
+
+### Model Architecture
+
+There are two types of models used to represent resources: proxy models and ES6 class models. In both cases, the models apply properties and methods to the resource, which defines how the resource can function in the UI and what other components can do with it. Different APIs return models in different structures, but the implementation of the models allows some common functionality to be available for any of them, usch as `someModel.name`, `someModel.description`, `setLabels` or `setAnnotations`.
```

Comment:
There's a minor typo here "usch as `someModel.name`, `someModel.description`..."
Otherwise all of this looks good to me, thanks for taking the time to write it up.

Reply:
updated